src/http/one/Tokenizer.cc

```diff
@@ -61,12 +43,14 @@ Http::One::Tokenizer::qdText(SBuf &returnedToken, const bool http1p0)
     // best we can do is a conditional reference since http1p0 value may change per-client
     const CharacterSet &tokenChars = (http1p0 ? qdtext1p0 : qdtext1p1);
 
-    for (;;) {
-        SBuf::size_type prefixLen = buf().findFirstNotOf(tokenChars);
-        returnedToken.append(consume(prefixLen));
+    while (!tok.atEnd()) {
+        if (tok.skip('"'))
+            return true;
 
-        // HTTP/1.1 allows quoted-pair, HTTP/1.0 does not
-        if (!http1p0 && skip('\\')) {
+        SBuf qdText;
+        if (tok.prefix(qdText, tokenChars))
+            returnedToken.append(qdText);
+        else if (!http1p0 && tok.skip('\\')) { // HTTP/1.1 allows quoted-pair, HTTP/1.0 does not
```

Comment:
Remove `else` to simplify the overall logic -- we can handle the quoted pair after handling the plain token prefix above AFAICT, without the need for another loop iteration.